lib/src/main/kotlin/com/swmansion/starknet/data/types/Responses.kt

```diff
@@ -41,18 +44,44 @@ data class DeployAccountResponse(
     val address: Felt? = null,
 )
 
-@OptIn(ExperimentalSerializationApi::class)
 @Serializable
 data class EstimateFeeResponse(
-    @JsonNames("gas_consumed", "gas_usage")
+    @SerialName("gas_consumed")
     val gasConsumed: Felt,
 
-    @JsonNames("gas_price")
+    @SerialName("gas_price")
     val gasPrice: Felt,
 
-    @JsonNames("overall_fee")
+    @SerialName("overall_fee")
     val overallFee: Felt,
-)
+
+    // TODO: (#344) Deviation from the spec, make this non-nullable once Pathfinder is updated
+    @SerialName("unit")
+    val feeUnit: PriceUnit? = null,
+) {
+    fun toMaxFee(overhead: Double = 0.5): Felt {
+        return addOverhead(overallFee.value, overhead).toFelt
+    }
+
+    fun toResourceBounds(
+        amountOverhead: Double = 0.1,
+        unitPriceOverhead: Double = 0.5,
+    ): ResourceBoundsMapping {
+        val maxAmount = addOverhead(gasConsumed.value, amountOverhead).toUint64
+        val maxPricePerUnit = addOverhead(gasPrice.value, unitPriceOverhead).toUint128
+
+        // As of Starknet 0.13.0, the L2 gas is not supported
+        // Because of this, the L2 gas values are hardcoded to 0
+        return ResourceBoundsMapping(
+            l1Gas = ResourceBounds(maxAmount = maxAmount, maxPricePerUnit = maxPricePerUnit),
+            l2Gas = ResourceBounds(maxAmount = Uint64.ZERO, maxPricePerUnit = Uint128.ZERO),
+        )
+    }
+    private fun addOverhead(value: BigInteger, overhead: Double): BigInteger {
+        val multiplier = ((1 + overhead) * 100).roundToInt().toBigInteger()
+        return value.multiply(multiplier).divide(BigInteger.valueOf(100))
+    }
```

Comment:
We typically don't define much methods inside files with just dataclasses.
However, I think the way it was done (just plain function), is sub-optimal, since we shouldn't really get max fee or resource bounds without having estimate response.

We could:
- move this particular response to separate file
- define those methods as extensions

Reply:
let's make an issue and think about it at later date

Reply:
Sure, #375